projects/igniteui-angular/src/lib/query-builder/query-builder.component.ts

```diff
@@ -768,10 +768,26 @@ export class IgxQueryBuilderComponent implements AfterViewInit, OnDestroy {
      */
     public deleteGroup() {
         const selectedGroup = this.contextualGroup;
```

Comment:
In order to avoid repetition, we could change the logic here to smth like:
```suggestion
       let selectedGroup = this.contextualGroup;
        let parent = selectedGroup.parent;
        if (!parent)
            this.rootGroup = null;

        while (parent) {
            let index = parent.children.indexOf(selectedGroup);
            parent.children.splice(index, 1);                     
            selectedGroup = parent;
            parent = parent.children.length === 0 ? parent.parent : null;
        }

        if (this.rootGroup?.children.length === 0) {
            this.rootGroup = null;
        }
```